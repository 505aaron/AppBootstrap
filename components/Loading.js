import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import { View, ActivityIndicator, StyleSheet, Text } from "react-native";

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center"
  }
});

class Loading extends PureComponent {
  static propTypes = {};

  render() {
    return (
      <View style={styles.container}>
        <ActivityIndicator />
        <Text>Loading...</Text>
      </View>
    );
  }
}

export default Loading;
