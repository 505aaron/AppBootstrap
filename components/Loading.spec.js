import React from 'react';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import Loading from './Loading';

describe('Loading', () => {
  let shallowRenderer;
  let baseProps;

  beforeEach(() => {
    baseProps = {};

    shallowRenderer = new ShallowRenderer();
  });

  test('render', () => {
    shallowRenderer.render(
      <Loading {...baseProps} />,
    );

    const tree = shallowRenderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
