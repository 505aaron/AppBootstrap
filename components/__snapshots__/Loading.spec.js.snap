// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Loading...
  </Text>
</View>
`;
